src/main/java/org/jfrog/buildinfo/deployment/BuildInfoRecorder.java

```diff
@@ -48,12 +49,14 @@
     private final ArtifactoryClientConfiguration conf;
     private final BuildDeployer buildDeployer;
     private final Log logger;
+    private final ExecutionListener wrappedListener;
 
-    public BuildInfoRecorder(MavenSession session, Log logger, ArtifactoryClientConfiguration conf) {
+    public BuildInfoRecorder(MavenSession session, Log logger, ArtifactoryClientConfiguration conf, ExecutionListener wrappedListener) {
         this.buildInfoBuilder = new BuildInfoModelPropertyResolver(logger, session, conf);
         this.buildDeployer = new BuildDeployer(logger);
         this.logger = logger;
         this.conf = conf;
+        this.wrappedListener = wrappedListener;
```

Comment:
I think using this logic will help us to remove 16 times of `if (wrappedListener != null) {`:
```suggestion
        this.wrappedListener = ObjectUtils.defaultIfNull(wrappedListener, new AbstractExecutionListener());
```

Reply:
I updated the code to use this `ObjectUtils.defaultIfNull` approach and made the changes to all the `wrappedListener` calls